layer/contracts/datasets.py

```diff
@@ -71,7 +71,6 @@ def __init__(
         self._version_id = version_id
         self.schema = schema
         self.metadata = metadata if metadata is not None else {}
-        self.build = build or DatasetBuild()
```

Comment:
Why do we pass a `build` to the constructor then?

Reply:
I believe this is a remnant from an old piece of code that is no longer used